### DISCUSSION TITLE
Data Cleansing pada Mulltiple Regression Model

### DISCUSSION THREAD

**Discussion by mosesnugraha** [teamalgoritma/community#588] [Q&A] [answered]
selamat Siang, saya mau tanya, untuk data cleansing pada multiple regression model, apabila semua variable predictor time datanya integer, dan mau mengubah tipe data ini apakah untuk salah satu variable atau bisa lebih satu?, yang saya baca dari materi RM mengubah tipe data yang tadinya int menjadi factor apakah harus memi
<img width="1440" alt="Screen Shot 2023-03-08 at 12 25 12" src="https://user-images.githubusercontent.com/123171973/223627101-d040a926-00c4-4974-8e72-b8a983995af8.png">
liki nilai uniquenya sedikit? dan apakah lebih dari satu predictor variable ini yang memiliki unique sedikit seperti 0,1 atau 1,0 ini harus semua dijadikan factor?, contoh dari data saya ada floor, indian marble, black marble , dan floors memiliki nilai unique yang sedikit. Apakah prediktor variable yang saya sebutkan ini harus diubah tipe datanya menjadi factor? Karena yang diterangkan dikelas itu, Hanna satu prediktor variable yang diubah tipe datanya menjadi factor. Contohnya ada digambar ini

**Comment by idoali** [comment, marked as answer]
Terima kasih pak atas pertanyaannya

Pada kasus regresi, yang diubah menjadi factor itu adalah kolom yang : 
- nilainya bukan angka
- nilainya diskret (tidak mengandung koma) dan tidak terlalu banyak jenisnya

Pada kasus yang bapak hadapi, kolom `Marble`, `Indian.Marble` dan `Floor` sudah berupa angka. Selain itu kemungkinan nilainya juga hanya dua, 0 dan 1. Maka kolom-kolom ini tidak perlu diubah menjadi factor.  

Semoga jawabannya cukup membantu pak
